### ISSUE TITLE
Provide ability to set scopeSelector and source options for configs

### ISSUE THREAD

**Issue by geksilla** [dvcrn/proton#103]
Currently we cannot set `scopeSelector` and `source` options via `proton.lib.atom.set-config!` see [Config::set](https://atom.io/docs/api/v1.3.3/Config#instance-set) for details.

**Comment by dvcrn** [comment]
How about change the API for configs from 
`[key value]` to `[key value opts]`. 
```
    ["core.themes" ["nucleus-dark-ui" "zenburn"]]
```

could turn into
```
    ["core.themes" ["nucleus-dark-ui" "zenburn"] {:scope ["foo"]}]
```

But what do we do with proton configs that can't get assigned to a scope? 

For the cljs code I think this is straight forward. We overload
```
(defn set-config! [selector value]
  (console! (str "Setting " selector " to " (clj->js value)))
  (.set config selector (clj->js value)))
```

To include a third parameter. I'll see if I can scribble something up.

**Comment by geksilla** [comment]
I also went with this approach, but since we convert configs to hashmap [here](https://github.com/dvcrn/proton/blob/master/src/cljs/proton/core.cljs#L82) we will lost third element `{:scope ["foo"]}` from vector.

**Comment by geksilla** [comment]
Fixed in #169
